homeassistant/helpers/check_config.py

```diff
@@ -48,22 +48,38 @@ def __init__(self) -> None:
         """Initialize HA config."""
         super().__init__()
         self.errors: list[CheckConfigError] = []
+        self.warnings: list[CheckConfigError] = []
 
     def add_error(
         self,
         message: str,
         domain: str | None = None,
         config: ConfigType | None = None,
     ) -> Self:
-        """Add a single error."""
+        """Add an error."""
         self.errors.append(CheckConfigError(str(message), domain, config))
         return self
 
     @property
     def error_str(self) -> str:
-        """Return errors as a string."""
+        """Concatenate all errors to a string."""
         return "\n".join([err.message for err in self.errors])
 
+    def add_warning(
+        self,
+        message: str,
+        domain: str | None = None,
+        config: ConfigType | None = None,
+    ) -> Self:
+        """Add a warning."""
+        self.warnings.append(CheckConfigError(str(message), domain, config))
+        return self
+
+    @property
+    def warning_str(self) -> str:
```

Comment:
Where is this used? Is it only used in tests?

Reply:
No, it's sent to frontend: https://github.com/home-assistant/core/pull/102902/files#diff-65868fe0aa58e2d13dd9b4cbaa58546997594d1150d398546c74123dc012eb7cR42